lib/common.js

```diff
@@ -183,23 +183,19 @@ function stringifyRequest(options, body) {
 }
 
 function isContentEncoded(headers) {
-  const contentEncoding = _.get(headers, 'content-encoding')
+  const contentEncoding = headers['content-encoding']
   return _.isString(contentEncoding) && contentEncoding !== ''
 }
 
 function contentEncoding(headers, encoder) {
-  const contentEncoding = _.get(headers, 'content-encoding')
+  const contentEncoding = headers['content-encoding']
   return contentEncoding === encoder
 }
 
 function isJSONContent(headers) {
-  let contentType = _.get(headers, 'content-type')
-  if (Array.isArray(contentType)) {
-    contentType = contentType[0]
-  }
-  contentType = (contentType || '').toLocaleLowerCase()
-
-  return contentType === 'application/json'
+  // https://tools.ietf.org/html/rfc8259
+  const contentType = (headers['content-type'] || '').toLowerCase()
+  return contentType.startsWith('application/json')
```

Comment:
Using `startsWith` vs `===` is actually parity with existing behavior. The main change here is that Node defines the `content-type` headers as a string not an array when there are dups. If you check out the old version, it would check if the value was an array and if so it would grab the first value. 

As to the legitimacy of having a charset on a JSON Content-Type header, it is a debate that will never die and I'm not sure Nock should be opinionated on the matter. For example, I know Expressjs always tacks on a charset for JSON responses because of a sec vun they had a few years ago.

Reply:
> As to the legitimacy of having a charset on a JSON Content-Type header, it is a debate that will never die and I'm not sure Nock should be opinionated on the matter. For example, I know Expressjs always tacks on a charset for JSON responses because of a sec vun they had a few years ago.

Right, I think my goal is similarly, not to have an opinion. In general we should do the standard thing rather than implement workarounds for edge cases. It's sometimes not obvious how to do that 😀